common/payload.cpp

```diff
@@ -60,27 +61,48 @@ namespace Hekate {
             "sdmc:/sept/payload.bin",
         };
 
-        bool LoadPayload() {
-            for (auto path : PayloadPaths) {
-                /* Clear payload buffer. */
-                std::memset(g_reboot_payload, 0xFF, sizeof(g_reboot_payload));
+        constexpr char const *const PayloadDirs[] = {
+            "sdmc:/",
+            "sdmc:/bootloader/payloads/",
+            "sdmc:/payloads/",
+        };
 
-                /* Open potential hekate payload. */
-                auto file = fopen(path, "r");
-                if (file == nullptr)
-                    continue;
+        bool LoadPayload(const char* path, bool hekate) {
+            /* Clear payload buffer. */
+            std::memset(g_reboot_payload, 0xFF, sizeof(g_reboot_payload));
 
-                /* Read payload to buffer. */
-                std::size_t ret = fread(g_reboot_payload, 1, sizeof(g_reboot_payload), file);
+            /* Open payload. */
+            auto file = fopen(path, "r");
+            if (file == nullptr)
+                return false;
 
-                /* Close file. */
-                fclose(file);
+            /* Read payload to buffer. */
+            std::size_t ret = fread(g_reboot_payload, 1, sizeof(g_reboot_payload), file);
 
-                /* Verify hekate payload loaded successfully. */
-                if (ret == 0 || *(u32 *)(g_reboot_payload + Hekate::MagicOffset) != Hekate::Magic)
-                    continue;
+            /* Close file. */
+            fclose(file);
+
+            /* Verify hekate payload loaded successfully. */
+            if(hekate) {
+                if (ret == 0 || *(u32 *)(g_reboot_payload + Payload::MagicOffset) != Payload::Magic)
+                    return false;
+            }
+
+            /* Verify payload loaded successfully. */
+            else{
+                if(ret == 0)
+                    return false;
+            }
```

Comment:
Newline and curly bracket style. Also try to avoid code duplication.
How about:
```c
/* Verify payload loaded successfully. */
if (ret == 0)
    return false;

/* Verify payload has hekate magic */
if (hekate && *(u32 *)(g_reboot_payload + Payload::MagicOffset) != Payload::Magic)
    return false;
```